src/tape/ensemble.py

```diff
@@ -37,8 +52,16 @@ def __init__(self, client=True, **kwargs):
         self._source = None  # Source Table
         self._object = None  # Object Table
 
-        self._source_dirty = False  # Source Dirty Flag
-        self._object_dirty = False  # Object Dirty Flag
+        self.frames = {}  # Frames managed by this Ensemble, keyed by label
+
+        # A unique ID to allocate new result frame labels.
+        self.default_frame_id = 1
+
+        self.source = None  # Source Table EnsembleFrame
```

Comment:
do we need both this and self._source?

Reply:
Yeah I was planning on removing post-merge so that we initially maintain full backwards compatibility, but I think this is fine to break and makes this PR cleaner.

Removed self._source and self._object in favor of just self.source and self.object